src/ui/public/selectable_row/selectable_row.js

```diff
@@ -0,0 +1,22 @@
+import { uiModules } from 'ui/modules';
+import template from './selectable_row.html';
+
+const app = uiModules.get('kibana');
+
+app.directive('selectableRow', function () {
+  return {
+    restrict: 'E',
+    replace: true,
+    transclude: true,
+    template: template,
+    scope: {
+      id: '=',
+      isSelected: '=',
+      onSelectChange: '=',
+    },
+    controllerAs: 'selectableRow',
```

Comment:
You may have already settled on a convention about this, but my 2c are that this may not be necessary. IMO the template is just as readable (if not slightly more so) like this:

```html
<input
  type="checkbox"
  class="kuiCheckBox"
  ng-click="this.onSelectChange(this.id, this.isSelected)"
  ng-model="this.isSelected"
>
```